java/src/org/openqa/selenium/remote/http/RetryRequest.java

```diff
@@ -54,7 +54,7 @@
       .handle(TimeoutException.class)
       .withBackoff(1, 4, ChronoUnit.SECONDS)
       .withMaxRetries(3)
-      .withMaxDuration(Duration.ofSeconds(10))
+      .withMaxDuration(Duration.ofMinutes(10))
```

Comment:
This seems like a long Read Timeout, any reason 10 minutes? Is it configurable?

Reply:
The retry mechanism uses two parameters to retry: maxRetries and maxDuration. Max duration is the tricky one to understand. Please refer to https://github.com/failsafe-lib/failsafe/pull/244/files to understand how to max duration actually works. We have a read timeout of 3 minutes and we want to do 3 retries, which would take 9 minutes since the error would be only thrown and hence retried if it exceeds 3 minutes. To round it off, it is set to 10 minutes. The main purpose is to allow proper retries. I agree that it should be configurable. That is the next step since it does require moving a few things around. Apologies, I have misunderstood the retry concepts here. I have earlier misunderstood the maxDuration setting, I earlier thought it was the total duration between two consecutive retries.

Reply:
Ah, yes, I think I misread this. Thanks.

Reply:
Setting the maximum duration isn't required. If it's not set, it will assume an "unlimited" max duration, which is actually limited by the max retries and the configured read timeout. Removing the `withMaxDuration` would resolve the configurability, if I decide to increase the timeout to be longer than 3 minutes, e.g. 5 minutes, the max duration would be 15 minutes and this would conflict with the 10 minutes here.

I think it meant to be used that you either specify a max duration or max attempts, so you retry until the max duration is reached or retry a maximum number of n times, from the code:
https://github.com/failsafe-lib/failsafe/blob/66b8ca3189f058c6938db04dc153e4746cb0141a/src/main/java/dev/failsafe/internal/RetryPolicyExecutor.java#L237

Reply:
That makes sense. If the user changes the timeout to a higher value, then this 10 minutes hard-coded value won't work. Setting only `withMaxRetries` should be fine.

Reply:
Thank you for pointing that out. Just relying on the maximum number of retries in our case is a good idea. I have made the desired changes.